sqlglot/parser.py

```diff
@@ -3153,7 +3162,28 @@ def _parse_ordered(self, parse_method: t.Optional[t.Callable] = None) -> exp.Ord
         ):
             nulls_first = True
 
-        return self.expression(exp.Ordered, this=this, desc=desc, nulls_first=nulls_first)
+        if self._match_text_seq("WITH", "FILL"):
+            from_ = None
+            to = None
+            step = None
+
+            if self._match(TokenType.FROM):
+                from_ = self._parse_bitwise()
+            if self._match_text_seq("TO"):
+                to = self._parse_bitwise()
+            if self._match_text_seq("STEP"):
+                step = self._parse_bitwise()
+
+            with_fill = self.expression(
+                exp.WithFill,
+                **{"from": from_, "to": to, "step": step},  # type: ignore
```

Comment:
just put it all inline? 

match(From) and parse_bitwise() then you don't have to declare